### PR TITLE
Add MetadataProducerFactory for predicates

### PR DESCRIPTION
**Pull request by alculquicondor**
/kind cleanup
/kind feature

**What this PR does / why we need it**:

This adds a factory for `predicates.MetadataProducer`, matching the stack for priorities.
This is necessary to configure the metadata producer, such as when adding defaults.

Also shortened names from `predicates.PredicateMetadata` to `predicates.Metadata` and similarly for priorities.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/priority important-soon

### CODE REVIEW

**Comment by alculquicondor** [comment]
/assign @kow3ns 
for pkg/controller/daemon

File: pkg/scheduler/factory.go [hunk 1]
```diff
@@ -539,8 +539,8 @@ func (c *Configurator) getPriorityConfigs(priorityKeys sets.String) ([]prioritie
 
 // GetPredicateMetadataProducer returns a function to build Predicate Metadata.
 // It is used by the scheduler and other components, such as k8s.io/autoscaler/cluster-autoscaler.
-func (c *Configurator) GetPredicateMetadataProducer() (predicates.PredicateMetadataProducer, error) {
-	return getPredicateMetadataProducer()
+func (c *Configurator) GetPredicateMetadataProducer() (predicates.MetadataProducer, error) {
+	return getPredicateMetadataProducer(*c.factoryArgs)
```
Comment (ahg-g) [inline pkg/scheduler/factory.go]: is this derefrecning safe?

Comment (ahg-g) [reply pkg/scheduler/factory.go]: also, you can get rid of this function completely, CA are moving to scheduler.New in 1.17

Comment (alculquicondor) [reply pkg/scheduler/factory.go]: It looks like there was no reason for it to be pointer anyways.
Function removed :)

**Review by janetkuo — APPROVED** [review]

**Pull request merged**